pystac_monty/sources/gfd.py

```diff
@@ -0,0 +1,204 @@
+
+        bbox = self._get_bounding_box(data["system:footprint"]["coordinates"])
+        # Episode number not in the source, so, set it to 1
+        episode_number = 1
+
+        startdate = pytz.utc.localize(datetime.fromtimestamp(data["system:time_start"] / 1000))
+        enddate = pytz.utc.localize(datetime.fromtimestamp(data["system:time_end"] / 1000))
+
+        item = Item(
+            id=f'{STAC_EVENT_ID_PREFIX}{data["system:index"]}',
+            geometry=geometry,
+            bbox=bbox,
+            datetime=startdate,
+            properties={
+                "title": description,
+                "description": description,
+                "start_datetime": startdate.isoformat(),
+                "end_datetime": enddate.isoformat(),
+            },
+        )
+
+        item.set_collection(self.get_event_collection())
+        item.properties["roles"] = ["source", "event"]
+
+        MontyExtension.add_to(item)
+        monty = MontyExtension.ext(item)
+        monty.episode_number = episode_number
+        monty.country_codes = data["cc"].split(",")
+        monty.hazard_codes = ["FL"]  # GFD is a Flood related source
+        monty.compute_and_set_correlation_id(hazard_profiles=self.hazard_profiles)
+
+        return item
+
+    def make_hazard_items(self) -> List[Item]:
+        """Create hazard items"""
+        hazard_items = []
+
+        gfd_data = self.check_and_get_gfd_data()
+        event_items = self.make_source_event_items()
+
+        for event_item, src_data in zip(event_items, gfd_data):
+            hazard_item = event_item.clone()
+            hazard_item.id = event_item.id.replace(STAC_EVENT_ID_PREFIX, STAC_HAZARD_ID_PREFIX)
+            hazard_item.properties["roles"] = ["source", "hazard"]
+            hazard_item.set_collection(self.get_hazard_collection())
+
+            monty = MontyExtension.ext(hazard_item)
+            # Hazard Detail
+            monty.hazard_detail = HazardDetail(
+                cluster=self.hazard_profiles.get_cluster_code(hazard_item),
+                severity_value=src_data["dfo_severity"],
+                severity_unit="GFD Flood Severity Score",
+                severity_label=None,
+                estimate_type=MontyEstimateType.PRIMARY,
+            )
+            hazard_items.append(hazard_item)
+        return hazard_items
+
+    def make_impact_items(self) -> List[Item]:
+        """Create impact items"""
+        items = []
+        gfd_data = self.check_and_get_gfd_data()
+        event_items = self.make_source_event_items()
+        for event_item, src_data in zip(event_items, gfd_data):
+            items.extend(self.make_type_based_impact_items(event_item, src_data))
+        return items
+
+    def make_type_based_impact_items(self, event_item: dict, src_data: dict) -> List[Item]:
+        """Returns the impact details related to flood"""
+        impact_fields = {
+            "dfo_dead": (MontyImpactExposureCategory.ALL_PEOPLE, MontyImpactType.DEATH),
```

Comment:
is `dfo_` prefix necessary here?

Reply:
@emmanuelmathot I used this as the key as this key is also present in the source response. This way, we can get the value directly from the response using this key without needing to have another mappings.